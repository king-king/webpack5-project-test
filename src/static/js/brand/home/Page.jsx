import React from 'react';
import Inuput1 from './Input1';

class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            value: 1
        };
    }
    shouldComponentUpdate() {
        return true;
    }
    select = value => {
        this.setState({ value });
    };
    render() {
        const in1 = <Inuput1 key='input-1' className='input-1' />;
        const in2 = (
            <div>
                {' '}
                <Inuput1 key='input-2' className='input-2' />
                {' '}
            </div>
        );
        return (
            <div>
                <div>
                    <button onClick={() => this.select(1)}>选择1</button>
                    <button onClick={() => this.select(2)}>选择2</button>
                </div>
                <div>
                    {this.state.value === 1 ? in1 : in2}
                </div>
            </div>
        );
    }
}
App.propTypes = {
};
export default App;
