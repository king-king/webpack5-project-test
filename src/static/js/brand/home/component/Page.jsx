import React from 'react';
import { Select } from '@jd/jmtd';
import PPageFramework from '@jd/sz-brand-react-component/dist/PageFrameWork';
import commonUrl from '../config/commonConfig';

class App extends React.Component {
    constructor() {
        super();
        this.state = {
            value: '1'
        };
    }
    shouldComponentUpdate() {
        return true;
    }
    setValue = value => {
        this.setState({ value });
    };
    render() {
        const data = [
            {
                label: '选项1',
                value: '1'
            },

            {
                label: '选项2',
                value: '2'
            },

            {
                label: '选项3',
                value: '3',
                disabled: true
            },

            {
                label: '选项4',
                value: '4'
            },

            {
                label: '选项5',
                value: '5'
            }
        ];
        return (
            <PPageFramework pageParam={{}} menuCode='' urlConfig={commonUrl}>
                <div>
                    太棒了，哈哈哈
                    <Select data={data} value={this.state.value} onChange={this.setValue} />
                </div>
            </PPageFramework>
        );
    }
}

App.propTypes = {};

export default App;
