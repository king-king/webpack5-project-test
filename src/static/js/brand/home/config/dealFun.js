const envHost = window.location.hostname; // 应用运行环境host

const onlineHostArr = ['ppzh.jd.com']; // 线上品牌版host
const onlineBdpHostArr = ['ppzh.bdp.jd.com', 'ge.jd.com']; // 线上运营版host

const localHost = ['local.brand.jd.com']; // 本地host, 由于本地host不走网关，所以只有一个域名

const preHostArr = ['web.ppzh.jd.com', '233.ppzh.jd.com', '80.ppzh.jd.com', 'zylp.jd.com']; // 商智品牌版测试、预发环境host
const preBdpHostArr = ['bdp.ppzh.jd.com', '25.ppzh.jd.com']; // 商智运营版测试、预发环境host

// 若共建方自己的测试、预发环境想直接访问商智的网关
const gzHostArr = ['beta-fans.quality.jd.com']; // 共建方host想直接访问商智品牌版网关
const gzBdpHostArr = ['quality-kanban.jd.com']; // 共建方host想直接访问商智运营版网关

// 现在我终于明白了，我们的网关需要配置host
const FOCUS_GATEWAY = true;

let curVersion = ''; // 默认当前版本是品牌版（如果是本地环境想走线上，那就直接改这个值）

// 品牌版
if (onlineHostArr.includes(envHost) || preHostArr.includes(envHost) || gzHostArr.includes(envHost)) {
    curVersion = '';
}

// 运营版
if (onlineBdpHostArr.includes(envHost) || preBdpHostArr.includes(envHost) || gzBdpHostArr.includes(envHost)) {
    curVersion = 'Bdp';
}

/**
 *
 * 线上环境host配置，任何时候都不能更改！！！
 * 线上环境host配置，任何时候都不能更改！！！
 * 线上环境host配置，任何时候都不能更改！！！
 * */

// 品牌版一下配置还需确认2020-11-11注
const onlineGateway = {
    common: 'https://zhgateway.jd.com', // 线上公共接口网关
    commonBdp: 'http://szbrom.back.jd.com', // 线上运营版网关
    business: 'https://zhgateway.jd.com/geajax', // fanajax是共建方的业务网关id，共建方需要替换成自己的网关id
    businessBdp: 'http://szbrom.back.jd.com/geajax' // 业务运营版网关
};

// 测试、预发网关配置，商智的测试 & 预发的网关域名都是同一个域名
const preGateway = {
    common: '//yf-szgateway.jd.com', // 测试、预发公共接口网关
    commonBdp: '//dev-szbrom.back.jd.com',
    business: '//yf-szgateway.jd.com/geajax', // fanajax是共建方的业务网关id，共建方需要替换成自己的网关id
    businessBdp: '//dev-szbrom.back.jd.com/geajax' // 业务运营版网关
};

/** *
 * 1、线上不管配置了什么，走统一的线上网关
 * 2、本地环境，全局强制
 * 3、测试环境 、 预发环境
 * @param hostObj
 * @param urlConfig
 * @returns {{}}
 */
export default function dealFun(hostObj, urlConfig) {
    if (onlineHostArr.includes(envHost) || onlineBdpHostArr.includes(envHost)) {
        // 线上环境
        hostObj.devhost = onlineGateway[`${hostObj.type}${curVersion}`];
    } else if (localHost.includes(envHost)) {
        // 本地环境
        if (FOCUS_GATEWAY) {
            hostObj.devhost = '';
            // hostObj.devhost = preGateway[`${hostObj.type}${curVersion}`];
        }
    } else if (gzHostArr.includes(envHost) || gzBdpHostArr.includes(envHost)) {
        // 共建方自己的测试、预发环境
        if (FOCUS_GATEWAY) {
            hostObj.devhost = preGateway[`${hostObj.type}${curVersion}`];
        }
    } else if (preHostArr.includes(envHost) || preBdpHostArr.includes(envHost)) {
        // 商智测试、预发环境
        if (FOCUS_GATEWAY) {
            hostObj.devhost = preGateway[`${hostObj.type}${curVersion}`];
        }
    } else {
        // 托底方案，认为是线上环境
        hostObj.devhost = onlineGateway[`${hostObj.type}${curVersion}`];
    }

    const retObj = {};
    for (const key in urlConfig) {
        if (urlConfig.hasOwnProperty(key) === true) {
            retObj[key] = hostObj.devhost + urlConfig[key];
        }
    }

    return retObj;
}

// 导出当前系统版本
export const sysVersion = curVersion;
