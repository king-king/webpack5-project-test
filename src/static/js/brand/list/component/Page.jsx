import React from 'react';

class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
        };
    }
    shouldComponentUpdate() {
        return true;
    }
    render() {
        return (
           <div>hello word</div>
        );
    }
}
App.propTypes = {
};
export default App;
