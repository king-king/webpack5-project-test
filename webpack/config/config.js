/*
 * Created by king at 2022-1-24 11:18:00
 * Copyright (c) 2022
*/

const os = require('os');

const appMap = {
    Linux: 'google chrome', // linux
    Darwin: 'google chrome', // mac
    // eslint-disable-next-line
    Windows_NT: 'chrome' // windows
};
module.exports.app = appMap[os.type()];

module.exports.flag = 'brandweb';
