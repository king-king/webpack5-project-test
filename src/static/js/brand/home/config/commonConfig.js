import dealFun from './dealFun';

const commonConfigHost = {
    type: 'common', // FOCUS_GATEWAY为false时， 可以自己设置devHost值，自定义是否走网关
    devHost: '' //
};

const commonUrlConfig = {
    getCommonMenu: '/brand/common/getMenuTreeData.ajax',
    globalInfo: '/brand/common/newGetGlobalInfo.ajax',
    getBrandAndCategoryNew: '/brand/common/getBrandAndCategoryNew.ajax', // 头部组件调用，查询品牌、类目和供应商信息
    getAdTypeData: '/brand/common/getAdTypeData.ajax', // 头部组件调用 获取广告类型数据
    searchSku: '/brand/common/common/searchSku.ajax', // 头部组件调用
    searchMultipleSkuByIds: '/brand/common/common/searchMultipleSku.ajax', // 头部组件调用
    getContactsInfo: '/brand/common/getContactsInfo.ajax', // 判断联系人信息是否填写完整
    getServerTime: '/brand/common/getServerTime.ajax', // 获取服务器时间
    getSystemNotice: '/brand/common/getSystemNotice.ajax', // 获取系统公告与菜单公告
    getContentTypeData: '/brand/contentMarketing/common/getContentTypeData.ajax', // 头部组件调用
    getWatermark: '/brand/common/getWatermark.ajax', // 获取运营版水印接口
    getWindowNotice: '/brand/common/getWindowNotice.ajax', // 获取弹窗公告
    passportLogout: '/brand/common/passportLogout.ajax', // passport登出接口
    getBdpUserInfo: '/brand/common/getBdpUserInfo.ajax', // 运营版验证用户身份是否隐藏品牌类目多选
    getSelfRunShop: '/brand/common/getSelfRunShop.ajax'
};

const retUrlConfig = dealFun(commonConfigHost, commonUrlConfig);

export default retUrlConfig;
