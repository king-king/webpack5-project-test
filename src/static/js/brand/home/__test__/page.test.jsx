import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Page from '../component/Page';
import { commonConfig } from './config/page';

Enzyme.configure({ adapter: new Adapter() });

describe('Content test', () => {
    const wrapper = shallow(<Page {...commonConfig} />);
    test('快照', () => {
        expect(wrapper.render()).toMatchSnapshot();
    });

    test('交互', () => {
        // 点击重置按钮
        wrapper.find('button').simulate('click');
    });
});
