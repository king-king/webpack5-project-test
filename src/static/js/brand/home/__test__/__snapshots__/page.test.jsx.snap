// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content test 快照 1`] = `
<div>
  <div>
    header
  </div>
  hello word
  <div>
    null
  </div>
  <div>
    null
  </div>
  <button>
    click
  </button>
</div>
`;
